### PR TITLE
docs: add HOWTO for migrating integrations to required input dependencies

### PR DESCRIPTION
**Pull request by teresaromero**
## Summary

Adds a HOWTO guide for migrating integration packages from inline agent templates to `requires.input` dependencies, based on lessons from the [`elastic_package_registry`](https://github.com/elastic/integrations/pull/19719) migration to `prometheus_input`.

The guide covers manifest changes, dataset handling, variable overrides, policy/system test updates, and known platform gaps. It is linked from `dependency_management.md`.

## Test plan

- [ ] Review doc for accuracy against the `elastic_package_registry` reference PR
- [ ] Verify local links resolve correctly in GitHub preview

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: docs/howto/migrate_integration_required_input_dependency.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
+# HOWTO: Migrate an integration package to use required input dependencies
+
+> **Reference implementation:** [elastic/integrations#19719](https://github.com/elastic/integrations/pull/19719) — `elastic_package_registry` migration from a legacy inline `prometheus/metrics` agent template to `prometheus_input` v1.0.1.
+
+## Introduction
+
+Integration packages historically duplicated agent collector configuration in local `agent/stream/*.ybs` templates. **Integrations with required input dependencies** delegate generic collector configuration to a reusable **input package** and keep the integration focused on service-specific defaults, field mappings, ingest pipelines, and dashboards.
+
+At build time, `elastic-package build` fetches required input packages from the package registry, merges their agent templates and variable definitions into the integration, and rewrites `package:` references to the concrete input types Fleet expects. See [HOWTO: Enable dependency management](./dependency_management.md#integrations-with-required-packages-and-the-package-registry) for the general model.
```
Comment (mrodm) [inline docs/howto/migrate_integration_required_input_dependency.md:9]: This makes me think if we should add some kind of retry in the `elastic-package build` command since now there are more network requests to EPR.

At least this reference: 
https://github.com/elastic/integrations/blob/c15203839eaff07c13a44a5fc73d8462df31d6e8/.buildkite/scripts/common.sh#L906

This is used for testing and release pipeline. It should be checked if there is any other usage of `elastic-package build` in the pipelines or scripts (even with the `ELASTIC_PACKAGE_BIN` variable).

File: docs/howto/migrate_integration_required_input_dependency.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+```yaml
+metrics_path: {{metrics_path}}
+tags:
+{{#each tags as |tag i|}}
+    - {{tag}}
+{{/each}}
+```
+
+Everything else is merged from the `prometheus_input` template at build time.
```
Comment (mrodm) [inline docs/howto/migrate_integration_required_input_dependency.md]: Is this not merged by Fleet (Kibana) ?

IIRC `elastic-package` adds the agent template from the integration and all the agent templates from the input packages (dependencies).

File: docs/howto/migrate_integration_required_input_dependency.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+- Manually document the input dependency if `{{ inputDocs }}` does not render (see [Documentation gaps](#documentation-gaps)).
+- Add changelog entries: migration itself (`enhancement`), stack constraint bump (`enhancement`), field-mapping fixes (`bugfix`).
+
+### 8. Verify end-to-end
+
+```bash
+elastic-package build
+elastic-package test
+elastic-package test policy -v
+elastic-package test system
```
Comment (mrodm) [inline docs/howto/migrate_integration_required_input_dependency.md]: These two commands are already part of `elastic-package test`.

File: docs/howto/migrate_integration_required_input_dependency.md [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+- [ ] `dataset` explicitly set on the data stream manifest when it must differ from the input default
+- [ ] Local `stream.yml.hbs` contains only integration-owned template fragments
+- [ ] Variable overrides use data stream `vars` (not silent template hardcoding) unless documented as intentional
+- [ ] `_dev/test/config.yml` declares `requires` for local input package during development
+- [ ] Policy tests: default + overrides, expectations reviewed for dataset, overridden defaults, and sibling streams (`enabled: false` where required)
+- [ ] System tests pass with realistic service traffic where needed
+- [ ] Pipeline regression tests for edge cases found during migration
+- [ ] Changelog entries: migration, stack constraint, field-mapping fixes
+- [ ] Docs manually updated if `{{ inputDocs }}` is empty
+- [ ] Dashboards validated on the target stack version
```
Comment (mrodm) [inline docs/howto/migrate_integration_required_input_dependency.md:408]: I would add another check item here to validate how variables are shown in Fleet when the integration is installed to create the policy. Validating that variables shown in the UI should be there and they have a corresponding entry in the agent template too. For instance, there is a variable in the UI, but this variable has a hardcoded value in the final agent template to be used by the Elastic Agents.

File: docs/howto/migrate_integration_required_input_dependency.md (hunk 2 above)
Comment (teresaromero) [reply docs/howto/migrate_integration_required_input_dependency.md]: fixed. the bundle imports the template from the input, and indeed is Fleet the one that merges all

File: docs/howto/migrate_integration_required_input_dependency.md (hunk 1 above)
Comment (teresaromero) [reply docs/howto/migrate_integration_required_input_dependency.md:9]: opened an issue https://github.com/elastic/elastic-package/issues/3756

File: docs/howto/migrate_integration_required_input_dependency.md (hunk 4 above)
Comment (teresaromero) [reply docs/howto/migrate_integration_required_input_dependency.md:408]: i've added some improvements on the description to reflect this https://github.com/elastic/elastic-package/pull/3753/changes/89fcc4fa0ee96dbe1d2f4fa5d94a96aeb610235f

**Review by mrodm — APPROVED** [review]
